nautobot/ipam/tests/test_filters.py

```diff
@@ -222,215 +226,184 @@ class PrefixTestCase(FilterTestCases.FilterTestCase, FilterTestCases.TenancyFilt
+        self.assertEqual(self.filterset({"present_in_vrf_id": vrfs[1].pk}, self.queryset).qs.count(), 2)
+        self.assertEqual(self.filterset({"present_in_vrf": vrfs[0].rd}, self.queryset).qs.count(), 6)
+        self.assertEqual(self.filterset({"present_in_vrf": vrfs[1].rd}, self.queryset).qs.count(), 2)
 
     def test_region(self):
-        regions = Region.objects.all()[:2]
+        regions = (
+            Region.objects.create(name="region1"),
+            Region.objects.create(name="region2"),
+            Region.objects.create(name="region3"),
+        )
+        test_sites = (
+            Site.objects.create(name="site1", region=regions[0], status=Status.objects.get_for_model(Site).first()),
+            Site.objects.create(name="site2", region=regions[1], status=Status.objects.get_for_model(Site).first()),
+            Site.objects.create(name="site3", region=regions[2], status=Status.objects.get_for_model(Site).first()),
+        )
+        PrefixFactory(site=test_sites[0])
+        PrefixFactory(site=test_sites[1])
         params = {"region_id": [regions[0].pk, regions[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+        self.assertQuerysetEqual(
+            self.filterset(params, self.queryset).qs, self.queryset.filter(site__region__in=params["region_id"])
+        )
         params = {"region": [regions[0].slug, regions[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+        self.assertQuerysetEqual(
+            self.filterset(params, self.queryset).qs, self.queryset.filter(site__region__slug__in=params["region"])
+        )
 
     def test_site(self):
-        sites = Site.objects.all()[:2]
-        params = {"site_id": [sites[0].pk, sites[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
-        params = {"site": [sites[0].slug, sites[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+        test_sites = (
+            Site.objects.create(name="site1", status=Status.objects.get_for_model(Site).first()),
+            Site.objects.create(name="site2", status=Status.objects.get_for_model(Site).first()),
+            Site.objects.create(name="site3", status=Status.objects.get_for_model(Site).first()),
+        )
+        PrefixFactory(site=test_sites[0])
+        PrefixFactory(site=test_sites[1])
+        params = {"site_id": [test_sites[0].pk, test_sites[1].pk]}
+        self.assertQuerysetEqual(
+            self.filterset(params, self.queryset).qs,
+            self.queryset.filter(site__in=[test_sites[0], test_sites[1]]),
+        )
+        params = {"site": [test_sites[0].slug, test_sites[1].slug]}
+        self.assertQuerysetEqual(
+            self.filterset(params, self.queryset).qs,
+            self.queryset.filter(site__in=[test_sites[0], test_sites[1]]),
+        )
```

Comment:
These will change/simplify once we have #2540, I presume?

Reply:
Yes. PrefixFactory is already setup to grab a random site when they're available.

Reply:
(Site and Location)